### ISSUE TITLE
[WPE-2.38] Seek is not working on Progressive MKV content.

### ISSUE THREAD

**Issue by suresh-khurdiya-epam** [Metrological/WebKitForWayland#1268]
Seek is not working on Progressive MKV content with WPE 2.38.
Seek tests are failed for all Progressive MKV content, first seek to the beginning (0s) is pass, but seek forward to 20s is always failed.

Only for MKV container, seek operation is failed. It seems like when it is trying for seek then it is coming to zero position due to that seek operation is failed.

So, basically, position is not correct during the seek operation for MKV based content. For other container like mp4, seek is working fine.

Any suggestion or leads?

**Comment by eocanha** [comment]
Debugged the behaviour of WebKitWebSrc on seeks. I was able to reproduce the issue in wpe-2.28, wpe-2.38, wpe-2.42 and also wpe-2.42 with the latest multimedia patches backported (very close to upstream), but not upstream. This last step gave me the clue that maybe the problem was in our custom buildroot patches. Indeed, [the 0003-matroskademux-Start-stream-time-at-zero.patch](https://github.com/WebPlatformForEmbedded/buildroot/blob/2145d5b7aeb1c079b8c5a497c5a566fcd98ce193/package/gstreamer1/gst1-plugins-good/1.18.6/0003-matroskademux-Start-stream-time-at-zero.patch) was causing the issue.

Tested the original "YouTube MSE 2018 20. VideoBufferSize" test without the patch on all those versions and it passed. Seeks on progressive video kept working as well. The issue happens in wpe-2.22 with and without the patch, and I couldn't check if test 20 would pass there (I would have needed to customize wpe to advertise that it supports vp9 on the Pi for that). Since wpe-2.22 is a legacy version, I'm not investing more time on it.

Pushed https://github.com/WebPlatformForEmbedded/buildroot/commit/fc0c21be3733b0a984a41d555b112cff610081ee to buildroot to only apply the patch to wpe-2.22.

Closing issue.
